shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -30,56 +32,156 @@ function _getLocalStorageObject(): Storage {
+                if (descriptor && descriptor.get) {
+                    canAccessStorage = true;
+                }
+            } catch (e) {
+                // If descriptor check fails, try direct access
+                // This will be caught by the outer try-catch if it fails
+                canAccessStorage = !!gbl[storageTypeName];
+            }
         }
-        let uid = (new Date).toString();
-        let storage: Storage = getGlobalInst(storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage");
-        let name:string = _storagePrefix + uid;
-        storage.setItem(name, uid);
-        let fail = storage.getItem(name) !== uid;
-        storage.removeItem(name);
-        if (!fail) {
-            return storage;
+        
+        // If we determined storage might be accessible, verify it works
+        if (canAccessStorage) {
+            try {
+                const uid = (new Date).toString();
+                const storage: Storage = getGlobalInst(storageTypeName);
+                const name = _storagePrefix + uid;
+                
+                storage.setItem(name, uid);
+                const success = storage.getItem(name) === uid;
+                storage.removeItem(name);
+                
+                if (success) {
+                    // Create a wrapped storage object that protects write operations
+                    const originalStorage = storage;
+                    const wrappedStorage = {
+                        // Read operations - don't reset cache on error
+                        getItem: function(key: string): string {
+                            try {
+                                return originalStorage.getItem(key);
+                            } catch (e) {
+                                return null;
+                            }
+                        },
+                        key: function(index: number): string {
+                            try {
+                                return originalStorage.key(index);
+                            } catch (e) {
+                                return null;
+                            }
+                        },
+                        get length(): number {
+                            try {
+                                return originalStorage.length;
+                            } catch (e) {
+                                return 0;
+                            }
+                        },
+                        
+                        // Write operations - reset cache on error
+                        setItem: function(key: string, value: string): void {
+                            try {
+                                originalStorage.setItem(key, value);
+                            } catch (e) {
+                                // Reset the cached storage on error
+                                if (storageType === StorageType.LocalStorage) {
```

Comment:
use the same approach as mentioned eariler, a function that returns the implementation - perhaps a single function that is passed a flag to determine if it should reset the storage variables